website/source/docs/auth/gcp.html.md

```diff
@@ -39,14 +39,15 @@ request to Vault. This helper is only available for IAM-type roles.
 ```text
 $ vault login -method=gcp \
     role="my-role" \
-    service_account="authenticating-account@my-project.iam.gserviceaccounts.com" \
+    service_account="authenticating-account@my-project.iam.gserviceaccount.com" \
     project="my-project" \
-    jwt_exp="15m" \
-    credentials=@path/to/signer/credentials.json
+    jwt_exp="15m"
 ```
 
 For more usage information, run `vault auth help gcp`.
 
+_NOTE:_ While it is possible to specify `credentials` in the login command, it is recommended to use the `GOOGLE_APPLICATION_CREDENTIALS` envrionment variable to point to the JSON file where credentials are stored.
```

Comment:
s/enviroment/environment

Reply:
I'm not sure this is actually recommended.

Reply:
Should I put it in the negative? ```
While it is possible to specify `credentials` in the login command it not is recommended.
``` if you run `vault auth help gcp` you'll see that specifying credentials is not recommended though I'm not entirely sure as to why since this key will be sent up in the payload regardless and there's a way to reference it without pasting it into the terminal.

```
 credentials=<string>			
	Explicitly specified GCP credentials in JSON string format (not recommended)
```

Just seems odd to have a `(not recommended)` text in the CLI help but have it be listed as part of the example in the web docs. I could also change this in the cli at https://github.com/hashicorp/vault-plugin-auth-gcp/blob/master/plugin/cli.go#L149 if it actually is ok I just think it would be good to have consistency is all.

@emilymye any thoughts?

Reply:
I think the _reason_ we say not to do it via the CLI is because it could show in history, but @emilymye can confirm

Reply:
FWIW I think it's a little odd that the env var points to a file but the `credentials` param is expected to be the actual JSON. Maybe it would be useful to have `credentials_file` or so?

Reply:
I agree, that would make things less confusing. Then we could update this code snippet to include `credentials_file` leaving `credentials` marked as `not recommended`

Reply:
Intended behavior: Use ADC instead of the `credentials` param. e.g. use implicit ADC if you running on GCE VM ideally, but you could also set them yourself. I think the `credentials` param is meant to be the JSON because you can specify the file using the `@` helper. Apologies, I wrote these docs when I didn't know GCP or Vault very well.

Reply:
@emilymye  or @jefferai would it be desired to have a credentials file attribute and phase out the raw credentials? Happy to make that change if it is :)

Reply:
I think it's fine to leave as is, since you can provide either raw credentials or the file (using "@") but I guess if we're very concerned about having the JSON float around, we can remove?

Reply:
For this use case, I think it makes sense to leave it here in some respect. My main confusion was that we say `not recommended` in the CLI help, but in the docs, it's part of the example. If I were to change anything it would be to remove the `(not recommended)` from the CLI docs and specify that it's only recommended to use with the `@` helper. But I'll defer to your judgment on it.